File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution {

  /**
    * @param ListNode $head
    * @param Integer $n
    * @return ListNode
    */
  function removeNthFromEnd($head, $n) {
    $node = $prev = $head = new ListNode(0, $head);
    while ($node->next) {
      $n ? $n-- : $prev = $prev->next;
      $node = $node->next;
    }
    $prev->next = $prev->next->next;
    return $head->next;
  }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
  def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
    head = ListNode(0, head)
    node, prev = head, head
    while node.next:
      if n:
        n -= 1
      else:
        prev = prev.next
      node = node.next
    if prev.next:
      prev.next = prev.next.next
    return head.next
        